File: AzurposService/app/src/main/java/merryweather/com/azurposservice/di/modules/AppModule.java
package merryweather.com.azurposservice.di.modules;

import android.app.Application;
import android.content.Context;


import dagger.Module;
import dagger.Provides;
import merryweather.com.azurposservice.di.qualifier.AppContext;

@Module
public class AppModule {

    private final Application application;

    public AppModule(Application application) {
        this.application = application;
    }

    @Provides
    @AppContext
    Context providesContext() {
        return application.getApplicationContext();
    }

}

File: Nista/app/src/main/java/com/example/nista/di/modules/AppModule.kt
package com.leroymerlin.cds.di.modules

import android.app.Application
import android.content.Context
import android.util.Log

import dagger.Module
import dagger.Provides
import com.leroymerlin.cds.di.qualifier.AppContext
import javax.inject.Singleton


@Module
class AppModule(private val application: Application) {

    @Provides
    @AppContext
    internal fun providesContext(): Context {
        return application.applicationContext
    }


}